sdk/identity/identity/src/credentials/interactiveBrowserCredential.ts

```diff
@@ -104,7 +109,37 @@ export class InteractiveBrowserCredential implements TokenCredential {
   ): Promise<AccessToken | null> {
     const scopeArray = typeof scopes === "object" ? scopes : [scopes];
 
-    return this.acquireTokenFromBrowser(scopeArray);
+    if (this.authenticationRecord && this.persistenceEnabled) {
+      return this.acquireTokenFromCache().catch((e) => {
+        if (e instanceof AuthenticationRequired) {
+          return this.acquireTokenFromBrowser(scopeArray);
+        } else {
+          throw e;
+        }
+      });
+    } else {
+      return this.acquireTokenFromBrowser(scopeArray);
+    }
+  }
+
+  private async acquireTokenFromCache(): Promise<AccessToken | null> {
+    await this.msalCacheManager.readFromPersistence();
+
+    const silentRequest = {
+      account: this.authenticationRecord!,
+      scopes: ["https://vault.azure.net/user_impersonation", "https://vault.azure.net/.default"]
+    };
+
+    try {
+      const response = await this.pca.acquireTokenSilent(silentRequest);
+      logger.info("Successful silent token acquisition");
+      return {
+        expiresOnTimestamp: response.expiresOn.getTime(),
+        token: response.accessToken
+      };
+    } catch (e) {
```

Comment:
I'm not sure that handling ___ALL___  exceptions thrown from `acquireTokenSilent` is the right thing to do here. Does MSAL document an particular error they will raise in the case that interaction is needed?

Reply:
From the MSAL docs:

"In case the refresh_token is expired or not found, an error is thrown and the guidance is for the user to call any interactive token acquisition API (eg: `acquireTokenByCode()`)."

We could ask them for more details, and see if they're comfortable locking down the exception side of the API or exposing more information.

Reply:
Yes, we should follow up with them on this. It would be nice to make the error contract for these cases a bit tighter.

Reply:
This sounds like it's out of scope for the time being. But they can look into it for a follow-up. I think we can take their change and modify what we do. It doesn't feel like necessarily a breaking change for us.